packages/api/src/logger/index.ts

```diff
@@ -0,0 +1,200 @@
+import pino, {
+  BaseLogger,
+  DestinationStream,
+  LevelWithSilent,
+  LoggerOptions,
+  redactOptions,
+} from 'pino'
+
+/**
+ * Determines if log environment is development or test
+ *
+ * Boolean
+ */
+const isDevelopment =
+  process.env.NODE_ENV === 'development' || process.env.NODE_ENV === 'test'
+
+/**
+ * Determines if log environment is production by checking if not development
+ *
+ * Boolean
+ */
+const isProduction = !isDevelopment
+
+/**
+ * Determines if logs should be prettified.
+ *
+ * Typically if logging to a transport stream or in production
+ * logs will not be prettified.
+ *
+ * In fact, the pino developers note:
+ * "We recommend against using pino-pretty in production,
+ * and highly recommend installing pino-pretty as a development dependency."
+ * See: https://github.com/pinojs/pino-pretty#programmatic-integration
+ *
+ * One exception to this rule may be Netlify functions logging.
+ * Its function logging output readability can benfeit from pretty-printing.
+ *
+ * Boolean
+ */
+export const isPretty = isDevelopment
+
+/**
+ * Defines the necessary pretty printing dependency
+ */
+export const prettifier = require('pino-pretty')
+
+/*
+ * List of keys to redact from log
+ *
+ * As an array, the redact option specifies paths that should have their values redacted from any log output.
+ *
+ * Each path must be a string using a syntax which corresponds to JavaScript dot and bracket notation.
+ *
+ * If an object is supplied, three options can be specified:
+ *
+ *      paths (String[]): Required. An array of paths
+ *      censor (String): Optional. A value to overwrite key which are to be redacted. Default: '[Redacted]'
+ *      remove (Boolean): Optional. Instead of censoring the value, remove both the key and the value. Default: false
+ */
+export const redactionsList: string[] | redactOptions = [
+  'host',
```

Comment:
Here are the docs about pino redaction:

https://github.com/pinojs/pino/blob/master/docs/redaction.md

In [path sytnax](https://github.com/pinojs/pino/blob/master/docs/redaction.md#path-syntax) it shows how to match keys and there are some wildcards.

But I don't think `host` would match `hosts` or `hosting` as key names. It doesn't redact based on key value.

>  if the defaults are good enough 

Right. I just made these up but I could see having these be a list of common PII like (in the US): 

`name, credit_card_number, ssn, phone_number, email, zip_code`

or even health info or other characteristics: `gender` etc.

Some sensible, default to be sure.

Reply:
@peterp @Tobbe or @jvanbaarsen Any thoughts or recommendations on a default set of redaction keys?

```js
export const redactionsList: string[] | redactOptions = [
  'host',
  'email',
  'accessToken',
  'access_token',
  'secret',
]
```

These seem sensible: `name, credit_card_number, ssn, phone_number, email, zip_code` but are heavily US-influenced.

Should I add directly or just suggest in docs/and or the `logger.ts` that gets added to one's app?

Reply:
Can we make this not only redact based on the key, but also on the value? I have two suggestions, but I'm sure there are more

1. Anything that looks like a jwt
2. Anything that is defined as an ENV var. Or, if that's too aggressive, how about anything that's defined as an ENV var with "KEY" or "TOKEN" in its name, and that isn't prefixed with REDWOOD_ENV_ and/or isn't included in "includeEnvironmentVariables"

Reply:
@Tobbe 

> Can we make this not only redact based on the key, but also on the value

Unfortunately, pino redacts on the key (or a path to the key).

See: https://github.com/pinojs/pino/blob/master/docs/redaction.md

> To redact sensitive information, supply paths to keys that hold sensitive data using the redact option:

I'll look into:

> Anything that is defined as an ENV var.

since those will be keys.

Reply:
I was actually thinking of the env var values.

I took inspiration from GitHub. They will scan the GH action workflow output for any string that matches the value of your secrets, or common encodings, like base64, and redact those.

So if you have a secret defined like `THREEPP_TOKEN=sdfsdf-osidfosi-sdfsis3sf-sdflsf` and it finds "sdfsdf-osidfosi-sdfsis3sf-sdflsf" anywhere in the output, it will hide that.

Reply:
Apparently redaction can be a very expensive operation (something you'd want your logger to avoid), thus `pino`uses [fast-redact](https://github.com/davidmarkclements/fast-redact) but it is also just key-level redaction.

I think it would have to check every key and then try to match on some regex for the values.

I like the idea and can see why super useful to filter out secrets or tokens.

Maybe an enhancement .. or ...

maybe add

'event.headers.authorization'
'headers.authorization'
'authorization'

That would remove the chance if you log the entire event that the bearer token is leaked?

Or what if you tried to log all `process.env` ? Maybe add

```
'DATABASE_URL'
'JWT_SECRET`
```
like that?

Reply:
Ohh, ok, so it's a performance thing. Yeah, def need the logs to be fast when running in prod. Unlike GH, we're (probably) not displaying the logs publicly. So it's not *as* critical if something does leak out.  Leave it with what you have, and then we can do this as an enhancement later if needed/wanted